### DISCUSSION TITLE
How to build a custom policy that is both proactive and reactive in nature? Or do they have to be 2 separate policies?

### DISCUSSION THREAD

**Discussion by bhehe** [App-vNext/Polly#1146] [Q&A]
I have a scenario where I'm evaluating building a custom policy so that the behavior can be managed pipeline/handler-style like the other Polly Policies we're using (i.e., Retry, Circuit Breaker, Bulkhead).

As I examined the custom policy templates from the Polly-Contrib repo I was trying to gauge whether or not it was feasible to build my two behaviors into a single policy as they are directly related and essentially handling the same concerns - one just happens to do something additional for retrying and evaluating whether to continue to retry or not/allow the failure to bubble out of the policy.

Any advice? I've started eyeballing the two templates side-by-side but being new to custom policies I'm wondering if this is the right thing to consider and can be done  versus just build two separate policies where one wraps the other.

**Comment by peter-csala** [comment]
Here are some hints for your trade-off analysis

|Policies|Pros|Cons|
|--|--|--|
|Wrap two policies|Polly's wrap functionality is battle hardened with reach functionalities (escalation, context propagation, CancellationToken propagation, etc.)|The policies are independent, Polly glues them together|
|Create a custom policy|The underlying policies can be more interdependent and you are not restricted to use unidirectional escalation | It is your responsibility to make it well-documented and tested, also future maintainers might require extra knowledge|
